SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc

```diff
@@ -79,67 +80,84 @@ void HGCPassive::update(const BeginOfRun * run) {
       auto it = (init_) ? mapLV_.find(plv) : findLV(plv);
+      double time   = aStep->GetTrack()->GetGlobalTime();
+      double energy = (aStep->GetTotalEnergyDeposit())/CLHEP::GeV;
+
+      unsigned int copy(0);
       if (((aStep->GetPostStepPoint() == nullptr) || 
 	   (aStep->GetTrack()->GetNextVolume() == nullptr)) &&
 	  (aStep->IsLastStepInVolume())) {
 #ifdef EDM_ML_DEBUG
-	std::cout << plv->GetName() << " F|L Step " 
-		  << aStep->IsFirstStepInVolume() << ":" 
-		  << aStep->IsLastStepInVolume() << " Position" 
-		  << aStep->GetPreStepPoint()->GetPosition() << " Track " 
-		  << aStep->GetTrack()->GetDefinition()->GetParticleName()
-		  << " at" << aStep->GetTrack()->GetPosition() << " Volume " 
-		  << aStep->GetTrack()->GetVolume() << ":" 
-		  << aStep->GetTrack()->GetNextVolume() << " Status " 
-		  << aStep->GetTrack()->GetTrackStatus() << " KE " 
-		  << aStep->GetTrack()->GetKineticEnergy() << " Deposit " 
-		  << aStep->GetTotalEnergyDeposit() << " Map "
-		  << (it != mapLV_.end()) << std::endl;
+	edm::LogVerbatim("ValidHGCal") << plv->GetName() << " F|L Step " 
+				       << aStep->IsFirstStepInVolume() << ":" 
+				       << aStep->IsLastStepInVolume() 
+				       << " Position" << aStep->GetPreStepPoint()->GetPosition()
+				       << " Track " << aStep->GetTrack()->GetDefinition()->GetParticleName()
+				       << " at" << aStep->GetTrack()->GetPosition() 
+				       << " Volume " << aStep->GetTrack()->GetVolume()
+				       << ":" << aStep->GetTrack()->GetNextVolume()
+				       << " Status " << aStep->GetTrack()->GetTrackStatus()
+				       << " KE " << aStep->GetTrack()->GetKineticEnergy()
+				       << " Deposit " << aStep->GetTotalEnergyDeposit()
+				       << " Map " << (it != mapLV_.end());
 #endif
-	double time   = aStep->GetTrack()->GetGlobalTime();
-	double energy = (aStep->GetPreStepPoint()->GetKineticEnergy() +
-			 aStep->GetTotalEnergyDeposit())/CLHEP::GeV;
-	if (it != mapLV_.end()) {
-	  storeInfo(it, plv, 0, time, energy);
-	} else if (topLV_ != nullptr) {
-	  auto itr = (init_) ? mapLV_.find(topLV_) : findLV(topLV_);
-	  if (itr != mapLV_.end()) {
-	    storeInfo(itr, topLV_, 0, time, energy);
-	  }
-	}
-      } else if (it != mapLV_.end()) {
-	unsigned int copy = (unsigned int)(touchable->GetReplicaNumber(0) + 
-					   1000*touchable->GetReplicaNumber(1));
-	double time = (aStep->GetPostStepPoint()->GetGlobalTime());
-	double edeposit = (aStep->GetTotalEnergyDeposit())/CLHEP::GeV;
-	storeInfo(it, plv, copy, time, edeposit);
+	energy += (aStep->GetPreStepPoint()->GetKineticEnergy()/CLHEP::GeV);
+      } else {
+	time = (aStep->GetPostStepPoint()->GetGlobalTime());
+	copy = (unsigned int)(touchable->GetReplicaNumber(0) + 
+			      1000*touchable->GetReplicaNumber(1));
+      }
+      if (it != mapLV_.end()) {
+	storeInfo(it, plv, copy, time, energy, true);
       } else if (topLV_ != nullptr) {
-	auto itr = findLV(topLV_);
+	auto itr = (init_) ? mapLV_.find(topLV_) : findLV(topLV_);
```

Comment:
hi @bsunanda - why is this change needed?

Reply:
Earlier it was in a different if-block. There is no need to do the test. I am putting it back in old sequence